src/harness/testcases/WINNF_FT_S_SIQ_testcase.py

```diff
@@ -67,6 +67,211 @@ def test_WINNF_FT_S_SIQ_2(self):
+        'highFrequency'] = 4200000000
+    fss['deploymentParam'][0]['installationParam']['latitude'] = 38.4
+    fss['deploymentParam'][0]['installationParam']['longitude'] = -98.9
+    self._sas_admin.InjectFss({'record': fss})
+
+    # Inject GWBL station operating at range 3650 - 3700 MHz and associated
+    # GWPZ, within 150 km of FSS (approx. 100 km from FSS location set above).
+    gwpz = json.load(
+        open(os.path.join('testcases', 'testdata', 'gwpz_record_0.json')))
+    self._sas_admin.InjectWisp(gwpz)
+
+    # Load PAL/PPA data, operating channel 3620 - 3630 MHz.
+    pal_record = json.load(
+        open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+    pal_low_frequency = 3620000000
+    pal_high_frequency = 3630000000
+    pal_user_id = 'pal_user'
+    ppa_record = json.load(
+        open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
+    ppa_record, pal_record = makePpaAndPalRecordsConsistent(ppa_record,
+                                                            [pal_record],
+                                                            pal_low_frequency,
+                                                            pal_high_frequency,
+                                                            pal_user_id)
+
+    # Load 5 CBSDs
+    device_1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    device_3 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_e.json')))
+    device_4 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_f.json')))
+    device_5 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_g.json')))
+
+    # Move device_1 inside PPA and outside 150kms range of FSS
+    device_1['installationParam']['latitude'] = 38.854
+    device_1['installationParam']['longitude'] = -97.195
+
+    # Move device_2 outside PPA and outside 150kms range of FSS
+    device_2['installationParam']['latitude'] = 40.0
+    device_2['installationParam']['longitude'] = -97.87
+
+    # Move device_3 inside PPA and inside 150kms range of FSS
+    device_3['installationParam']['latitude'] = 38.775
+    device_3['installationParam']['longitude'] = -97.291
+
+    # Move device_4 outside PPA and inside 150kms range of FSS
+    device_4['installationParam']['latitude'] = 39.0
+    device_4['installationParam']['longitude'] = -98.6
+
+    # Move device_5 outside PPA and outside 150kms range of FSS
+    device_5['installationParam']['latitude'] = 38.2
+    device_5['installationParam']['longitude'] = -99.5
+
+    # Register devices - Use same userId as in PAL record for devices part of
+    # the PPA cluster list.
+    device_1['userId'] = pal_user_id
+    device_3['userId'] = pal_user_id
+    devices = [device_1, device_2, device_3, device_4, device_5]
+    cbsd_ids = self.assertRegistered(devices)
```

Comment:
Test spec has CPAS trigger as step #4 before step #5 register five CBSDs.

Reply:
Test spec order for this testcase is a little bit misleading.
Devices 1 and 3 must be part of the cluster list of the PPA, so I need to add them to the PPA info before injecting zone data. For this, I need to register them to get their cbsdId. So actually steps 3 and 5 are connected and cannot be done separately. That's why step 4 is coming last.

Reply:
That makes sense. Thanks!